localeredirector/services/LocaleRedirectorService.php

```diff
@@ -74,15 +74,14 @@ public function getBrowserLanguageMatch()
   private function newUrl($locale)
   {
     $qs = $this->querystring ? '?' . $this->querystring : '';
-    $segments = craft()->request->segments;
-    $lastSegment = end($segments);
-    $criteria = craft()->elements->getCriteria(ElementType::Entry);
-    $criteria->slug = $lastSegment;
-    $entry = $criteria->first();
+    $element = craft()->urlManager->getMatchedElement();
```

Comment:
I discovered the **UrlManager** getMatchedElement method and it automatically gets the current element (entry) by url matching. https://craftcms.com/classreference/etc/web/UrlManager#getMatchedElement-detail So no more needing to check the last segment and trying to figure out a match on the slug, which singles don't have anyway.

Reply:
nice!